pkg/manifest/mantaray/walker_test.go

```diff
@@ -115,153 +114,23 @@ func TestWalkNode(t *testing.T) {
 			walkedCount := 0
 
 			walker := func(path []byte, node *mantaray.Node, err error) error {
-				walkedCount++
 
-				if !pathExists(path, tc.expected) {
-					return fmt.Errorf("walkFn returned unknown path: %s", path)
+				if !pathExistsInRightSequence(path, tc.expected, walkedCount) {
+					return fmt.Errorf("walkFn returned unexpected path: %s", path)
 				}
-
-				return nil
-			}
-			// Expect no errors.
-			err = n2.WalkNode(ctx, []byte{}, ls, walker)
-			if err != nil {
-				t.Fatalf("no error expected, found: %s", err)
-			}
-
-			if len(tc.expected) != walkedCount {
-				t.Errorf("expected %d nodes, got %d", len(tc.expected), walkedCount)
-			}
-		})
-	}
-}
-
-func TestWalk(t *testing.T) {
-	t.Parallel()
-
-	for _, tc := range []struct {
-		name     string
-		toAdd    [][]byte
-		expected [][]byte
-	}{
-		{
-			name: "simple",
-			toAdd: [][]byte{
-				[]byte("index.html"),
-				[]byte("img/test/"),
-				[]byte("img/test/oho.png"),
-				[]byte("img/test/old/test.png"),
-				// file with same prefix but not a directory prefix
-				[]byte("img/test/old/test.png.backup"),
```

Comment:
This test path is important to be tested in TestWalkNode function as it ensures that files that have the same prefix are treated correctly. Could you add to TestWalkNode file named "index.html.backup" in toAdd and expected slices so that this test case is preserved since TestWalk function is removed?

Reply:
Done. I extended the test with additional paths.

Reply:
Thank you.